app/Repositories/PostRepository.php

```diff
@@ -87,8 +87,8 @@ public function create(array $data, $signupId, $authenticatedUserRole = null)
             'action' => isset($data['action']) ? $data['action'] : null,
             'url' => $fileUrl,
             'text' => isset($data['text']) ? $data['text'] : null,
-            'status' => 'pending',
-            'source' => token()->client(),
+            'status' => isset($data['status']) ? $data['status'] : 'pending',
```

Comment:
I'm not sure we want to do this here. 

Down below on [lines 100 -> 104](https://github.com/DoSomething/rogue/blob/ee23e4dbd699ed45ae9526b78c37f4b7a8b1a46f/app/Repositories/PostRepository.php#L100-L104) I added some logic that checks the token scope or admin role and decides if it is allow to update those fields. 

If  request is coming from Rogue Admin that is using a `client_id=rogue-qa` (or whatever) then We check the scopes on that client token that says, yes you can update status and we will use your client id as the source. 

the client `dev-outh` should only be used on dev machines, so it's more likely that we want to use the `client_id` for source than not. For example, Chompy uses a `importer-client` id so then when we create posts (because that client has the right scopes) we set the status that is passed in the request and set the source as the `token()->client()`

Reply:
Yeah, we for sure only want this to be overridden by admins/machines. ➕

Reply:
ah yes this totally makes sense and didn't see the code below! It wasn't working before so that is why I had to update the `$isAdmin` line - I'll write more about it in the change!